### DISCUSSION TITLE
loadFromJson() reviver function getting called multiple times for each object

### DISCUSSION THREAD

**Discussion by sudiptokush** [fabricjs/fabric.js#9456] [Q&A]
I am working on Fabric JS in an Angular 16 application and I am seeing a strange behavior while trying to load JSON in a canvas. Each object in my JSON is causing the reviver function to be called multiple times. Unable to understand why. Any help is really appreciated. Following is my code.

```
temporaryCanvas.loadFromJSON(json, async () => {
  // logic here
  canvas.renderAll();
  canvas.onHistory();
}, (o: unknown, object: fabric.Object) => {
  console.log(o); // Each object getting printed twice hence had to check before pushing 
  if(allObjects.indexOf(object) === -1) allObjects.push(object);
});
```

JSON Structure
![image](https://github.com/fabricjs/fabric.js/assets/6592374/ddd5ecb5-b018-414f-9adf-7171358ccb37)

Output of the reviver console.log
![image](https://github.com/fabricjs/fabric.js/assets/6592374/bb7f880a-8231-475d-8814-9cf90991444f)

Note- If I load another image using loadFromJSON the reviver it gets called 4 times per object then 8 times and so on

**Comment by ShaMan123** [comment]
Angular has been reported to cause many wierd bugs because of their infrastructure.
There is somewhere in a related ticket a fix for that, not tracking the canvas in memory somehow. I really don't remember and never used Angular.
If you manage to reproduce without angular please open a bug report
